bindings/rust/wasmedge-sdk/src/vm.rs

```diff
@@ -468,6 +496,15 @@ impl Engine for Vm {
         Ok(returns)
     }
 }
+impl Clone for Vm {
+    fn clone(&self) -> Vm {
+        let config = self
+            .config_snapshot
+            .as_ref()
+            .map(|config| Config::copy_from(config).unwrap());
+        Vm::new(config).unwrap()
+    }
+}
```

Comment:
The `config_snapshot` field would introduce additional memory consumption if `clone` is never or rarely used in upstream apps; According to the code in `clone` function, in addition, it intends to create a new vm instance with a "copy" of `config`, so it does not exactly match the semantics of "Clone" trait. Based on these two reasons, it is recommended to use the following sample code to realize the idea of "cloning" a vm instead of implementing `Clone` trait for `Vm`:
```rust
// Assume that a config instance is created and named "config" already, then
let config_copy = Config::copy_from(&config)?;
// create a new vm instance with config_copy
let new_vm = Vm::new(Some(config_copy))?;
...
```

Reply:
I am afraid this is not possible. This behavior (clone and pass engine) is needed by  wasm-shim. I suggest you could help review whole runwasi project to design a reasonable clone function fit this use case. I think the code of shim is closed for modification at this moment.

Reply:
Could you please specify the relevant code snippets in `runwasi` project? I'll reference it to consider how to design an API for "clone". Thanks!

Reply:
https://github.com/second-state/runwasi/blob/408e4a009efde080b6ee4fef234632d4d6740448/crates/containerd-shim-wasm/src/sandbox/shim.rs#L923
I think there exist more use cases in other places. You could just mark clone function as comment and recompile it will see more detail error message.

run `make build` or `cargo test -- --nocapture` on top layer

Reply:
Ok, I see now. The `Local` struct requires the `E` must implement the `Clone` trait. Let's keep the clone implementation. Thanks!